graphics/opencv4/Portfile

```diff
@@ -15,24 +16,49 @@ license             BSD
 maintainers         {stromnov @stromnov} openmaintainer
 
 description         Open Source Computer Vision Library
-long_description    ${description}
 
-homepage            https://opencv.org
-master_sites        https://github.com/opencv/opencv/archive/:opencv
-distname            ${version}
-
-distfiles           ${distname}.tar.gz:opencv
+long_description    OpenCV (Open Source Computer Vision Library) is an\
+                    open source computer vision and machine learning\
+                    software library. OpenCV was built to provide a\
+                    common infrastructure for computer vision\
+                    applications and to accelerate the use of machine\
+                    perception in the commercial products. Being a\
+                    BSD-licensed product, OpenCV makes it easy for\
+                    businesses to utilize and modify the code.
 
-checksums           ${distname}.tar.gz \
-                    rmd160  0e23d239058c9a08bd4aa9c67977951bd4bc93b1 \
-                    sha256  5de5d96bdfb9dad6e6061d70f47a0a91cee96bb35afb9afb9ecb3d43e243d217 \
-                    size    88427411
+homepage            https://opencv.org
 
-extract.only        ${distname}.tar.gz
+conflicts           opencv
 
-worksrcdir          opencv-${version}
+master_sites        ${github.master_sites}:opencv
+
+distfiles           "[lindex ${distfiles} 0]:opencv"
+
+checksums           ${distname}${extract.suffix} \
+                    rmd160  81a44c106251695e65c6c9390568040569964883 \
+                    sha256  865a1fa5b4339ec3a70ec75dbdc013e8801a891685c5518342e95e045f8bbd08 \
+                    size    87807189
+
+# modify from _resources/port1.0/group/github-1.0.tcl to avoid file tag issue
+proc move_gh_repo {repo_dir_patt new_name} {
+    global distfiles workpath worksrcpath
+    if {(![file exists ${worksrcpath}] || ![file exists ${workpath}/${new_name}]) && \
+            [llength ${distfiles}] > 0 && \
+            [llength [glob -nocomplain ${workpath}/*]] > 0} {
+        if {[file exists [glob -nocomplain ${workpath}/${repo_dir_patt}]] && \
+            [file isdirectory [glob -nocomplain ${workpath}/${repo_dir_patt}]]} {
+            move [glob ${workpath}/${repo_dir_patt}] ${workpath}/${new_name}
+        } else {
+            # tarball is not "${repo_dir_patt}"
+            ui_error "\n\ngithub PortGroup: Error: \${worksrcpath} does not exist after extracting distfiles. This might indicate that the author or project is different than set in the Portfile due to a rename at GitHub. Please examine the extracted directory in ${workpath} and try to correct the Portfile by either changing the author or project or adding the worksrcdir option with the correct directory name.\n"
+            return -code error "Unexpected github tarball extract."
+        }
+    }
+}
```

Comment:
why is all of this required now - did keeping the ```extract_only``` and other statements in original Portfile not suffice? In other words, did you try the minimal changes (e.g., just a simple version update)?

Reply:
The `github-1.0.tcl` just does this the renaming on the main repo, but we need multiple repos, so I just used a code block from that portgroup to write this `proc`.